### PR TITLE
use proper method to hook channel handlers

### PR DESCRIPTION
**Pull request by florianreuth**
This pull request moves the adding of the packet_priority_handler, in 1.19.4 Mojang added a new method for adding pipeline handlers, this also indirectly fixes the incompatibility with ViaFabricPlus

### CODE REVIEW

File: src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java [hunk 1]
```diff
@@ -67,12 +62,13 @@ public class PacketPriorityHandler extends ChannelDuplexHandler {
...
     private static final int DEFAULT_SNDBUF = 8 * 1024;
 
-    public static void setupPacketPriority(Channel channel) {
+    public static void setupPacketPriority(ChannelPipeline pipeline) {
         if (Config.USE_PACKET_PRIORITY_SYSTEM) {
-            if (channel instanceof SocketChannel) {
-                channel.pipeline().addLast("vmp_packet_priority", new PacketPriorityHandler());
-                channel.config().setOption(ChannelOption.WRITE_BUFFER_WATER_MARK, new WriteBufferWaterMark(4 * 1024, 8 * 1024));
-                channel.config().setOption(ChannelOption.IP_TOS, IP_TOS_LOWDELAY | IP_TOS_THROUGHPUT); // reduce latency
+            if (pipeline instanceof SocketChannel) {
```
Comment (ishland) [inline src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java]: This obviously won't work as a `ChannelPipeline` is never going to be a `SocketChannel`

Use `pipeline.channel()` instead.

File: src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java [hunk 2]
```diff
@@ -67,12 +62,13 @@ public class PacketPriorityHandler extends ChannelDuplexHandler {
...
+                pipeline.addLast("vmp_packet_priority", new PacketPriorityHandler());
+                final ChannelConfig config = pipeline.channel().config();
```
Comment (ishland) [inline src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java]: Inline this variable.

**Review by ishland — CHANGES_REQUESTED** [review]
This PR looks good overall but there are some changes needed.

File: src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java (hunk 1 above)
Comment (florianreuth) [reply src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java]: done, sorry, forgot to change it

File: src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java (hunk 2 above)
Comment (florianreuth) [reply src/main/java/com/ishland/vmp/common/networking/priority/PacketPriorityHandler.java]: done

**Comment by florianreuth** [comment]
So is the PR fine now?

**Comment by toinouH** [comment]
Most probably but ishland is almost never available during the week.
So you might hear more about it on Friday or Saturday.

**Review by ishland — APPROVED** [review]

**Pull request merged**